test/dynamo/test_ctx_manager.py

```diff
@@ -949,9 +995,14 @@ def fn(a, b):
         self.assertTrue(res[0].dtype == torch.float16)
         self.assertTrue(res[1].dtype == torch.float16)
 
-    def test_generic_ctx_manager_with_graph_break(self):
+    @parametrize(
+        "Ctx",
+        [CustomizedCtxManagerWithGraphBreak, customized_ctx_manager_with_graph_break],
+        name_fn=lambda x: x.__name__,
+    )
+    def test_generic_ctx_manager_with_graph_break(self, Ctx):
         def fn(x):
-            with CustomizedCtxManagerWithGraphBreak(False):
+            with Ctx(False):
```

Comment:
This is a really weird test case. If the context maanger did nothing, then the output would still be the same. Can we do something like:

```
assert torch.is_grad_enabled()
x = torch.randn(2, 3, requires_grad=True)
opt_fn = torch.compile(backend="eager", fullgraph=False)(fn)
assert torch.is_grad_enabled()
self.assertFalse(opt_fn.requires_grad)  # since it was run under torch.no_grad.
```

Reply:
Or branch on is_grad_enabled like in the below tests.